src/overlays/actors/ovl_Bg_Spot06_Objects/z_bg_spot06_objects.c

```diff
@@ -29,39 +64,440 @@ const ActorInit Bg_Spot06_Objects_InitVars = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot06_Objects/func_808AEEFC.s")
+    if (this->timer == 0) {
+        this->dyna.actor.velocity.y = 0.5f;
+        this->dyna.actor.flags &= ~0x2000;
+
+        this->actionFunc = BgSpot06Objects_LockSwimToSurface;
+    }
+}
+
+/*
+ * After being pulled all the way out the fish shaped lock will rise to the surface, creating bubbles in the water as it
+ * does so.
+ */
+void BgSpot06Objects_LockSwimToSurface(BgSpot06Objects* this, GlobalContext* globalCtx) {
+    f32 cos;
+    f32 pad;
+
+    this->dyna.actor.posRot.pos.y += this->dyna.actor.velocity.y;
+
+    if (this->dyna.actor.velocity.y <= 0.0f) {
+        cos = Math_CosS(this->dyna.actor.shape.rot.x) * 4.3f;
+        this->dyna.actor.posRot.pos.x += (cos * Math_SinS(this->dyna.actor.shape.rot.y));
+        this->dyna.actor.posRot.pos.z += (cos * Math_CosS(this->dyna.actor.shape.rot.y));
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.posRot.pos.y - 1.3f;
+        BgSpot06Objects_LockSpawnWaterRipples(this, globalCtx, 0);
+
+        if (Math_ScaledStepToS(&this->dyna.actor.shape, 0, 0x260) != 0) {
+            this->dyna.actor.initPosRot.pos.x =
+                this->dyna.actor.posRot.pos.x - (Math_SinS(this->dyna.actor.shape.rot.y) * 16.0f);
+            this->dyna.actor.initPosRot.pos.z =
+                this->dyna.actor.posRot.pos.z - (Math_CosS(this->dyna.actor.shape.rot.y) * 16.0f);
+            this->dyna.actor.posRot.pos.y = -1993.0f;
+            this->timer = 32;
+            this->dyna.actor.flags &= ~0x10;
+            this->collider.list[0].dim.worldSphere.radius = this->collider.list[0].dim.modelSphere.radius * 2;
+            this->actionFunc = BgSpot06Objects_LockFloat;
+        }
+    } else {
+        if (this->dyna.actor.posRot.pos.y >= (-1973.0f)) {
+            this->dyna.actor.velocity.y = 0.0f;
+            BgSpot06Objects_LockSpawnWaterRipples(this, globalCtx, 1);
+            EffectSsGSplash_Spawn(globalCtx, &this->dyna.actor.initPosRot, NULL, NULL, 1, 700);
+        } else if (this->dyna.actor.shape.rot.x == -0x4000) {
+            this->dyna.actor.velocity.y += 0.02f;
+            this->dyna.actor.posRot.pos.x = Rand_CenteredFloat(1.0f) + this->dyna.actor.initPosRot.pos.x;
+            this->dyna.actor.posRot.pos.z = Rand_CenteredFloat(1.0f) + this->dyna.actor.initPosRot.pos.z;
+            this->dyna.actor.velocity.y =
+                (this->dyna.actor.velocity.y > 10.0f) ? (10.0f) : (this->dyna.actor.velocity.y);
+            BgSpot06Objects_LockSpawnBubbles(this, globalCtx, 0);
+        } else {
+            BgSpot06Objects_LockSpawnBubbles(this, globalCtx, 0);
+
+            if (Math_ScaledStepToS(&this->dyna.actor.shape.rot.x, -0x4000, 0x30)) {
+                this->dyna.actor.initPosRot.pos.x = this->dyna.actor.posRot.pos.x;
+                this->dyna.actor.initPosRot.pos.y = -1993.0f;
+                this->dyna.actor.initPosRot.pos.z = ((0, this->dyna)).actor.posRot.pos.z;
```

Comment:
This is definitely fake, is there no better solution? Maybe something suggested by the permuter?

Reply:
oh that's very strange I don't remember doing that, yeah I'll see what I can do